client/test-helpers/testing-library/index.js

```diff
@@ -31,3 +32,30 @@ export const renderWithProvider = (
 
 	return rtlRender( ui, { wrapper: Wrapper, ...renderOptions } );
 };
+
+export const renderHookWithProvider = (
```

Comment:
Interesting. 
I wish these helpers could be extracted and put into a finer package for reuse.

Reply:
Definitely a good candidate to package out 🤔 .

Reply:
Or not at all right now. Just a thought - didn't really expect a follow up 🙇 

This PR looks like in great shape AFAIC. Thanks for all the tests and follow ups.